packages/o11ytsdb/rust/src/lib.rs

```
@@ -1649,58 +1701,99 @@ fn decode_values_alp_inner(input: &[u8], val_out: &mut [f64]) -> usize {
-                for i in 0..n {
-                    val_out[i] = sortable_u64_to_f64(min_su64 + r.read_bits(exc_bw));
+                let mut prev = first_su64;
+                for i in 1..exc_count {
+                    let zz = r.read_bits(actual_bw);
+                    // Un-zigzag: (zz >> 1) ^ -(zz & 1)
+                    let d = ((zz >> 1) as i64) ^ (-((zz & 1) as i64));
+                    let cur = (prev as i128 + d as i128) as u64;
+                    exc_u64[i] = cur;
+                    prev = cur;
                 }
+                pos += ((exc_count - 1) * actual_bw as usize + 7) / 8;
```

Clippy (1.94.1)

[warning] value assigned to `pos` is never read

(warning)